### PR TITLE
#169817863 user account activation and email verification

### PR DESCRIPTION
**Pull request by shemaeric1234**
### What does this PR do?
An email should be sent to a user upon successful registration  and should be able to receive a verification link via email upon successful signup
### Description of Task to be completed?
- The user after sign-up he or she should receive an email a which includes a verification link
- User should activate his or her account by clicking on the verification link

### How should this be manually tested?
After clone repository goes to the root of the project open path into command run the npm install and once setup is done click npm test.

Using Postman to test endpoint:
Key: Content-Type value: application/JSON
To test authentication, add this to the header: Bearer TOKEN
Test endpoints
GET request
visit: URL localhost:3000/api/v1/activate/{token here}

### What are the relevant pivotal tracker stories?
[#169817863](https://www.pivotaltracker.com/story/show/169817863)

### Screenshots (if appropriate)
image
image

### CODE REVIEW

File: src/controllers/Users.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import UserServices from '../services/users/servises';
+import util from '../util/util';
+
+/**
+ * user class
+ */
+class UserController {
+  /**
+   * @param {int} req
```
Comment (aanchirinah) [inline src/controllers/Users.js]: ### Nitpick
- Add a descriptive JS Doc

File: src/database/migrations/20191205184718-create-users.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    lastname: {
+      type: Sequelize.STRING
+    },
+    email: {
+      type: Sequelize.STRING
+    },
+    password: {
+      type: Sequelize.STRING
+    },
+    active: {
```
Comment (aanchirinah) [inline src/database/migrations/20191205184718-create-users.js:26]: ### Nitpick

- Give a default value for this.

File: src/helpers/sendEmail.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        </div>
+       <a href="#" style="text-decoration: none; "> <div class="Email-wrapper_button" style="cursor: pointer; justify-self: center; margin-left: 80px;">Get Started</div></a>
+
+
+    </div>
+</body>
+</html>`;
+
+/**
+ * @param {string} to
```
Comment (aanchirinah) [inline src/helpers/sendEmail.js]: ### Nitpick
- Descriptive JS Doc.

File: src/middlewares/verifyToken.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+dotenv.config();
+
+const hasSiggned = (req, res, next) => {
+  if (req.headers.authorization === undefined) {
+    return res.status(401).send({
+      status: '401',
+      error: 'Please Set The Authorization Header!',
+    });
+  }
+  const token = req.headers.authorization.split(' ')[1];
```
Comment (aanchirinah) [inline src/middlewares/verifyToken.js]: ### Nitpick

- You may want to check if it's a `Bearer` token also.

File: src/services/users/servises.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+   * @param {int} id
+   * @param {object} updateUser
+   * @returns {object} return object
+   */
+  static async activeUser(id, updateUser) {
+    /**
+     *@return {object} return object
+     */
+
+    const userToUpdate = await database.users.findOne({
```
Comment (aanchirinah) [inline src/services/users/servises.js]: ### Nitpick

- You need to perform some sort of exception handling also.

File: src/tests/users.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    chai
+      .request(app)
+      .patch('/API/v2/activate/1*')
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('it should check if user accoutn is activated', (done) => {
```
Comment (aanchirinah) [inline src/tests/users.js]:
```suggestion
  it('it should check if user account is activated', (done) => {
```

**Review by aanchirinah — CHANGES_REQUESTED** [review]
Good job just kindly attend to my requests.

File: src/tests/users.js (hunk 6 above)
Comment (shemaeric1234) [reply src/tests/users.js]: I have done

**Review by jo-rdan — APPROVED** [review]
Eric good job so far!

**Comment by shemaeric1234** [comment]
I have completed the feedback , so you can review

**Review by aanchirinah — APPROVED** [review]

**Review by Jajabenit250 — APPROVED** [review]

**Pull request merged**